seerep_srv/seerep_server/src/fb_cameraintrinsics_service.cpp

```diff
@@ -68,6 +68,48 @@ grpc::Status FbCameraIntrinsicsService::TransferCameraIntrinsics(
 
   BOOST_LOG_SEV(m_logger, boost::log::trivial::severity_level::info) << "received camera instrinsics for storage ";
 
+  // check if the distortion model is found in kCameraDistortionModels
+  bool distortion_model_err =
+      !requestRoot->distortion_model() ||
+      std::find(std::begin(seerep_server_constants::kCameraDistortionModels),
+                std::end(seerep_server_constants::kCameraDistortionModels),
+                requestRoot->distortion_model()->c_str()) == std::end(seerep_server_constants::kCameraDistortionModels);
+
+  bool distortion_mat_err = !requestRoot->distortion() || requestRoot->distortion()->size() < 1;
+  bool intrinsics_mat_err = !requestRoot->intrinsic_matrix() || requestRoot->intrinsic_matrix()->size() != 9;
+  bool rectification_mat_err = !requestRoot->rectification_matrix() || requestRoot->rectification_matrix()->size() != 9;
+  bool projection_mat_err = !requestRoot->projection_matrix() || requestRoot->projection_matrix()->size() != 12;
```

Comment:
only the intrinsic mat is a must have. The rest may be empty if the information is unknown. The check should allow empty mat for these.

Reply:
Only allowing empty matrices causes another error while writing the values to hdf5 for pb at least. Maybe checking that a matrix has only one entry could signal that its values should be ignored.